transport/http/outbound.go

```diff
@@ -111,30 +152,35 @@ func (o outbound) Call(ctx context.Context, req *transport.Request) (*transport.
 	deadline, _ := ctx.Deadline()
 	ttl := deadline.Sub(start)
 
-	request, err := http.NewRequest("POST", o.URL, req.Body)
+	req, err := http.NewRequest("POST", o.URL, treq.Body)
 	if err != nil {
 		return nil, err
 	}
 
-	request.Header = applicationHeaders.ToHTTPHeaders(req.Headers, nil)
-	if hs := baggage.FromContext(ctx); hs.Len() > 0 {
-		request.Header = baggageHeaders.ToHTTPHeaders(hs, request.Header)
-	}
+	ctx, span := o.createSpan(ctx, req, treq, start)
+	defer span.Finish()
 
-	request.Header.Set(CallerHeader, req.Caller)
-	request.Header.Set(ServiceHeader, req.Service)
-	request.Header.Set(ProcedureHeader, req.Procedure)
-	request.Header.Set(TTLMSHeader, fmt.Sprintf("%d", ttl/time.Millisecond))
+	req.Header.Set(CallerHeader, treq.Caller)
+	req.Header.Set(ServiceHeader, treq.Service)
+	req.Header.Set(ProcedureHeader, treq.Procedure)
+	req.Header.Set(TTLMSHeader, fmt.Sprintf("%d", ttl/time.Millisecond))
 
-	encoding := string(req.Encoding)
+	encoding := string(treq.Encoding)
 	if encoding != "" {
-		request.Header.Set(EncodingHeader, encoding)
+		req.Header.Set(EncodingHeader, encoding)
+	}
+
+	response, err := ctxhttp.Do(ctx, o.Client, req)
+
+	if response != nil {
```

Comment:
you can safely assume that you have a status code `if err != nil`,

From the docs: https://golang.org/pkg/net/http/#Client.Do
```
An error is returned if caused by client policy (such as CheckRedirect), or failure to speak HTTP (such as a network connectivity problem). A non-2xx status code doesn't cause an error.
```

I would remove this check, and have it set right before line 189 where we check the response code